plugins/Unity/Launcher/xdgwatcher.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2019 UBports project.
+ * Author(s): Marius Gripsgard <marius@ubports.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "xdgwatcher.h"
+
+#include <QDebug>
+#include <QDir>
+#include <QFile>
+#include <QFileInfo>
+#include <QStandardPaths>
+#include <QTextStream>
+
+XdgWatcher::XdgWatcher(QObject* parent)
+    : QObject(parent),
+      m_watcher(new QFileSystemWatcher(this))
+{
+    connect(m_watcher, &QFileSystemWatcher::directoryChanged, this, &XdgWatcher::onDirectoryChanged);
+    connect(m_watcher, &QFileSystemWatcher::fileChanged, this, &XdgWatcher::onFileChanged);
+
+    const auto paths = QStandardPaths::standardLocations(QStandardPaths::ApplicationsLocation);
```

Comment:
As this will only watch directories in the host, this PR doesn't solve the issue for install/remove of apps in Libertine. How are we going to handle those?

Reply:
So the idea to that was to make symlinks the .desktop to the standard locations instead, this makes it easier for libertine work on other DE (if needed)